flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobKey.java

```diff
@@ -83,30 +113,34 @@ public boolean equals(final Object obj) {
 
 		final BlobKey bk = (BlobKey) obj;
 
-		return Arrays.equals(this.key, bk.key);
+		return Arrays.equals(this.key, bk.key) && this.type == bk.type;
 	}
 
 	@Override
 	public int hashCode() {
-		return Arrays.hashCode(this.key);
+		return Arrays.hashCode(this.key) + this.type.hashCode();
```

Comment:
I think when one combines field, then one should multiply the fields with a prime, e.g. `37`.

Reply:
ok, I wasn't quite sure and based my code on some examples from the current Flink code. Apparently, I took the wrong ones.